app/code/local/Shipfunk/Shipfunk/Model/Carrier.php

```diff
@@ -78,8 +78,8 @@ public function _getQuotes($postCode)
                     "city" => (!empty($request->getDestCity()))?urlencode($request->getDestCity()):"Temp city",
                     "country" => (!empty($request->getDestCountryId()))?$request->getDestCountryId():"FI",
                     "phone" => (!empty($request->getTelephone()))?urlencode($request->getTelephone()):"1234567890",
-                    "email" => (!empty($customer->getEmail()) && !filter_var($customer->getEmail(), FILTER_VALIDATE_EMAIL) === false)?urlencode($customer->getEmail()):"shipfunk@shipfunk.fi"
-                ) 
+                    "email" => (!empty($customer->getEmail()) && !filter_var($customer->getEmail(), FILTER_VALIDATE_EMAIL) === false)?$customer->getEmail():"shipfunk@shipfunk.fi"
```

Comment:
I would argue that defaulting the customers email into something that may or may not even exist, is not a proper way to go. I don't know where that one came in the first place, but let's get rid of it entirely.

```suggestion
                    "email" => (!empty($customer->getEmail()) && !filter_var($customer->getEmail(), FILTER_VALIDATE_EMAIL) === false)?$customer->getEmail():null
```